R/dvhydrograph-data.R

```diff
@@ -126,17 +138,28 @@ parseDVMinMaxIVs <- function(reportObject, timezone, type, invertedFlag, exclude
 #' vertical lines connecting the steps between those TS.
 #' @param stat the parsed non-estimated time series
 #' @param est the parsed estimated time series
+#' @param excludeZeroNegativeFlag whether or not zero / negative values
+#' will be removed
 #' @return a list of vertical lines connecting steps between stat and est
 #' @importFrom dplyr arrange
-getEstimatedEdges <- function(stat, est){
+#' @return a list of the edges with the folowing columns time, y0, y1,
+#' and newSet. Time is the x position of the edge, y0 is the first y
+#' value and y1 is the second. newSet is what the next time series is
+getEstimatedEdges <- function(stat, est, excludeZeroNegativeFlag=NULL){
   estEdges <- list()
 
-  if(isEmptyOrBlank(est[['value']]) || isEmptyOrBlank(stat[['value']])){
+  if(is.null(stat) || is.null(est) || isEmptyOrBlank(est[['points']][['value']]) || isEmptyOrBlank(stat[['points']][['value']])){
     return(NULL)
   }
-  
-  est <- est[c('time', 'value')]
-  stat <- stat[c('time', 'value')]
+
+  #Don't render edges for values that will be removed if we are exculding zero and negative values
+  if(!is.null(excludeZeroNegativeFlag) && excludeZeroNegativeFlag){
```

Comment:
I think we want to get away from just using `is.null` and using `isEmptyOrBlank` to capture the multitude of things it might come through as.